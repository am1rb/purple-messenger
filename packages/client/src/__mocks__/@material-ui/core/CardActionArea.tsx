import React from "react";
import type { CardActionAreaProps } from "@material-ui/core/CardActionArea";

function MuiCardActionArea({ children }: CardActionAreaProps) {
  return <div data-testid="mock-mui-card-action-area">{children}</div>;
}

export default MuiCardActionArea;
