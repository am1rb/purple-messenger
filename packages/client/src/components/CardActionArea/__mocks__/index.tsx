import React from "react";

interface CardActionAreaProps {
  children: React.ReactNode;
}

function CardActionArea({ children }: CardActionAreaProps) {
  return <div data-testid="mock-card-action-area">{children}</div>;
}

export default CardActionArea;
