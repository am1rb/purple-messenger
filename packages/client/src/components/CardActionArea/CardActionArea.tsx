import { CardActionArea } from "@material-ui/core";
import withButtonLink from "components/withButtonLink";

export default withButtonLink(CardActionArea);
