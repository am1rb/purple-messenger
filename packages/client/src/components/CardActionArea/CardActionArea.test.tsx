import React from "react";
import { render } from "@testing-library/react";
import CardActionArea from "./CardActionArea";

jest.mock("components/withButtonLink");

describe("CardActionArea Component Tests", () => {
  it("Should match the snapshot", () => {
    const { container } = render(
      <CardActionArea>custom children</CardActionArea>
    );
    expect(container).toMatchSnapshot();
  });
});
