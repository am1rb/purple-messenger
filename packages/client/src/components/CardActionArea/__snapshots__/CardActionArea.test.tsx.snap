// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardActionArea Component Tests Should match the snapshot 1`] = `
<div>
  <div
    data-testid="mock-mui-card-action-area"
  >
    custom children
  </div>
</div>
`;
